app/models/spree/gtpay_transaction.rb

```diff
@@ -58,13 +56,16 @@ def update_transaction(transaction_params)
       if successful_status_code?
         update_transaction_on_query
       else
-        self.save
+        self.save(validate: false)
```

Comment:
This is because irrespective of status, the record of the transaction must be saved in the database